sources/common/inc/VcdExceptionList.h

```diff
@@ -0,0 +1,80 @@
+/// @file VcdExceptionList.h
+///
+/// The file lists errors and warnings which might occur within vcdMaker
+/// applications.
+///
+/// @par Full Description
+/// vcdMaker errors and warnings.
+///
+/// @ingroup Exception
+///
+/// @par Copyright (c) 2017 vcdMaker team
+///
+/// Permission is hereby granted, free of charge, to any person obtaining a
+/// copy of this software and associated documentation files (the "Software"),
+/// to deal in the Software without restriction, including without limitation
+/// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+/// and/or sell copies of the Software, and to permit persons to whom the
+/// Software is furnished to do so, subject to the following conditions:
+///
+/// The above copyright notice and this permission notice shall be included
+/// in all copies or substantial portions of the Software.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+/// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+/// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+/// IN THE SOFTWARE.
+
+#pragma once
+
+namespace EXCEPTION
```

Comment:
Well... I think we just need a basic logger that takes some code and message. Message can be generated ad-hoc or created using helper function in class that is interested in generating a message. (See my general comment on the top) Codes can be just defined in particular class or module level. I'm not a big fan of any codes btw, message should be informative enough so that no code referring to documentation should be needed. Error message can just directly point to doc chapter if this is required, but most of the time it shouldn't be. Just say what is wrong in the message. And logger shouldn't decide when execution must be stopped. It should just log a message... Error (in any form - exception, return value) should be propagated to a place in which we can decide what to do now.